modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/util/KBArticleLockManagerUtil.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.knowledge.base.web.internal.util;
+
+import com.liferay.knowledge.base.constants.KBArticleConstants;
+import com.liferay.knowledge.base.model.KBArticle;
+import com.liferay.petra.function.UnsafeFunction;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.feature.flag.FeatureFlagManagerUtil;
+import com.liferay.portal.kernel.lock.LockManagerUtil;
+
+/**
+ * @author Marco Galluzzi
+ */
+public class KBArticleLockManagerUtil {
```

Comment:
1) I've put all the logic in one class to avoid repeating code, including FF checks, and be sure to call the correct version of `lock` and `unlock` methods from `LockManagerUtil` class (not all of them do the same 😟 ).

2) Creating a Util class was the easiest, and independent, option. Moving the code to the service is what I'm asking in the description because I'm not sure if it is the way to go. But this can be done in a following PR

Reply:
I think yes, It should be inside the services.

From my perspective Lock and unlock operations should be part of the business logic and should be handled in this layer (Service) and not in the Web layer.

For example, it is how it is used in other services, as `com.liferay.portlet.documentlibrary.service.impl.DLFileEntryLocalServiceImpl#lockFileEntry` or `com.liferay.portlet.documentlibrary.service.impl.DLFileEntryLocalServiceImpl#unlockFileEntry`

In this case, instead of creating a new util you could create a similar method in the service to lock and unlock KBArticles (and have the FF there)

Reply:
I will change it, thanks for your feedback.

Reply:
👍